yb-voyager/src/queryparser/unsupportedConstructs.go

```diff
@@ -0,0 +1,126 @@
+package queryparser
+
+import (
+	"slices"
+
+	pg_query "github.com/pganalyze/pg_query_go/v5"
+)
+
+const (
+	ADVISORY_LOCKS = "Advisory Locks"
+	SYSTEM_COLUMNS = "System Columns"
+	XML_FUNCTIONS  = "XML Functions"
+)
+
+// NOTE: pg parser converts the func names in parse tree to lower case by default
+var advisoryLockFunctions = []string{
+	"pg_advisory_lock", "pg_try_advisory_lock", "pg_advisory_xact_lock",
+	"pg_advisory_unlock", "pg_advisory_unlock_all", "pg_try_advisory_xact_lock",
+}
+
+func (qp *QueryParser) containsAdvisoryLocks() bool {
+	if qp.ParseTree == nil {
+		return false
+	}
+
+	selectStmtNode, isSelectStmt := qp.ParseTree.Stmts[0].Stmt.Node.(*pg_query.Node_SelectStmt)
+	if !isSelectStmt {
+		return false
+	}
+
+	// Check advisory locks in the target list
+	if containsAdvisoryLocksInTargetList(selectStmtNode.SelectStmt.TargetList) {
+		return true
+	}
+
+	// Check advisory locks in FROM clause
+	if containsAdvisoryLocksInFromClause(selectStmtNode.SelectStmt.FromClause) {
+		return true
+	}
+
+	// Check advisory locks in WHERE clause
+	if containsAdvisoryLocksInWhereClause(selectStmtNode.SelectStmt.WhereClause) {
+		return true
+	}
+	return false
+}
+
+// Checks for advisory lock functions in the main query's target list.
+// This includes direct function calls like:
+// Example: SELECT pg_advisory_lock(1);
+// Example: SELECT col1, pg_try_advisory_lock(2) FROM my_table;
```

Comment:
Can you please add the parse tree for all of these different ways in comments, it will help to review this part of code.